proxy_agent/src/main.rs

```diff
@@ -72,22 +86,47 @@ fn main() {
     } else {
         #[cfg(windows)]
         {
-            _ = service_dispatcher::start(constants::PROXY_AGENT_SERVICE_NAME, ffi_service_main);
+            //
+            let main = std::thread::spawn(|| {
+                _ = service_dispatcher::start(
+                    constants::PROXY_AGENT_SERVICE_NAME,
+                    ffi_service_main,
+                );
+            });
+
+            main.join().unwrap();
         }
 
         #[cfg(not(windows))]
         {
-            service::start_service_wait();
+            service::start_service_wait().await;
         }
     }
 }
 
 #[cfg(windows)]
-fn proxy_agent_windows_service_main(args: Vec<OsString>) {
+fn proxy_agent_windows_service_main(_args: Vec<OsString>) {
     // start the Instant to calculate the elapsed time
     _ = helpers::get_elapsed_time_in_millisec();
 
-    if let Err(e) = windows::run_service(args) {
-        logger::write_error(format!("{e}"));
+    match ASYNC_RUNTIME_HANDLE.get() {
+        Some(handle) => {
+            handle.spawn(windows::run_service());
```

Comment:
Is the Windows service lifetime tied to `proxy_agent_windows_service_main`? In other words, do you want this call to block until `run_service` completes? If so you want [block_on](https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html#method.block_on).

Reply:
we need block_on